### PR TITLE
Throw appropriate NameNotRecognizedException when unknown identifier is found

### PR DESCRIPTION
**Pull request by igitur**
We were throwing inappropriate exceptions for a cell formula e.g. `=XXX`.

### CODE REVIEW

**Comment by igitur** [comment]
Solved issue at https://dev.to/masanori_msl/net-core-closedxml-getting-cell-values-1-p3o

**Comment by igitur** [comment]
I should update the test names too.

File: ClosedXML_Tests/Excel/CalcEngine/MathTrigTests.cs [hunk 1]
```diff
@@ -955,19 +947,18 @@ public void Floor(double input, double significance, double expectedResult)
...
         public void Floor_ThrowsDivisionByZeroOnZeroSignificance(double input, double significance)
         {
             Assert.Throws<DivisionByZeroException>(() => XLWorkbook.EvaluateExpr($"FLOOR({input.ToInvariantString()}, {significance.ToInvariantString()})"));
         }
 
+        [TestCase(6.7, -1)]
+        public void Floor_ThrowsNumberExceptionOnInvalidInput(double input, double significance)
+        {
+            Assert.Throws<NumberException>(() => XLWorkbook.EvaluateExpr($"FLOOR({input.ToInvariantString()}, {significance.ToInvariantString()})"));
+        }
```
Comment (Pankraty) [inline ClosedXML_Tests/Excel/CalcEngine/MathTrigTests.cs:961]: Here and after `Atan2_ReturnsMinus3QuartersOfPiWhenFirstSmaller0AndSecondItsNegative` the empty lines between methods are missing.

**Review by Pankraty — APPROVED** [review]
Found minor formatting issues. Otherwise is fine.

**Pull request merged**